src/odoc.ml

```diff
@@ -9,116 +9,156 @@ let ( ++ ) = Path.relative
 let get_odoc sctx = SC.resolve_program sctx "odoc" ~hint:"opam install odoc"
 let odoc_ext = ".odoc"
 
-let module_deps (m : Module.t) ~dir ~dep_graph ~modules =
+module Mld : sig
+  type t
+  val create : name:string -> t
+
+  val odoc_file : doc_dir:Path.t -> t -> Path.t
+  val odoc_input : doc_dir:Path.t -> t -> Path.t
```

Comment:
I can remove this refactoring if necessary. It proved to be useful for experimenting with renaming things, but now it's purpose seems dubious. It still doesn't seem so bad to keep this stuff behind an interface so I'm leaning to keep it.